### DISCUSSION TITLE
Invalid file when importing a route from WhatsApp

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#651] [Q&A] [answered]
Till now it worked to share routes between devices or friends in Kurviger format via the menu Routes --> Share/Export --> Share.
I used this easy way very often. You could import the file without any change in the other device and save it with the identical name. This doesn't work any more. When the kurviger file from Whatsapp is now tapped and opened with Cruiser, it tries to open it as GPX instead of Kurviger file and results in the error message "invalid file".

Shared with V5.0.1 b11 via Whatsapp

![Unbenannt](https://github.com/user-attachments/assets/ea38c49a-68a0-4b1e-8ae9-ad60ebf92efd)

Imported again with V5.0.1 b11: it isn't recognized as kurviger file, but as gpx file
![Unbenannt](https://github.com/user-attachments/assets/2eadd791-302b-427c-b685-6138599f6571)

and results in this error
![Unbenannt](https://github.com/user-attachments/assets/b2ad6309-cb7c-4f0e-9208-7c68789a0d2f)

When I do the same with V4, it imports without any issues.

**Comment by devemux86** [comment]
@DolomitesTom Thanks for the report.

As I do not use WhatsApp, this is difficult to reproduce, unless you can also see it with another app we all use.

I will check the code to see if something can be improved.

**Reply by DolomitesTom** [reply]
> unless you can also see it with another app we all use.

Which messenger for example? Threema? I do not use it anymore, but could reinstall it.

**Reply by devemux86** [reply]
Google Messages?

**Reply by DolomitesTom** [reply]
Have never used it, just installed, but only wants to send messages via SMS/MMs and I don't have a SIM card in my old Phone anymore. It does not send via WLAN, even if this option is active.

**Reply by devemux86** [reply]
@DolomitesTom 

How do you compare Cruiser 5 and 4? On different phones with different Androids from different OEMs?
Or the comparison is made on the same device?

WhatsApp could work differently in the 2 different Androids / devices.

**Reply by DolomitesTom** [reply]
The test is with the same device, same OS, Android 10 on Samsung S8 plus. I share the file to myself, this is possible in WhatsApp, unfortunately not in threema, so I cannot test it there. When I re open the file from the WhatsApp message on the same device with V5, the error occurs. When I open the same file (as WhatsApp message) on the other device with V4, Samsung A52 5G with Android 14, it opens with Cruiser directly without any problem. Can the Json format be connected manually by default directly with Cruiser? Because with this format it says unknown format.

**Reply by devemux86** [reply]
@DolomitesTom Can you try **5.0.0** from Google Play?

**Reply by DolomitesTom** [reply]
Sure, how can I restore this? Just says open (=b11) or deinstall, but not reinstall. Could you provide a download link for the apk?

**Reply by DolomitesTom** [reply]
I found an older Version 5.0.0 b1, but it couldn't be installed, there was an error message something like this. I have deinstalled it now, but have to download all maps and pois again. This takes a while. Don't want to do this too often 🙈.

**Comment by DolomitesTom** [comment]
The problem also occurs with Google play version 5.0.0

I didn't use V5 too often for the known reasons and only started to use it on my old device recently. I want to give it a try.

That's probably the reason why I didn't discover this before.

**Reply by devemux86** [reply]
Doesn't the device have Google Play? Or you can download it from the [Releases](https://github.com/devemux86/cruiser/releases).

If it really happens with 5.0.0 then it is difficult to check it, as I cannot reproduce it.
And the import process was not changed.

**Comment by devemux86** [comment]
@DolomitesTom Can you reproduce it with a route attached in email?

**Reply by DolomitesTom** [reply]
No, via E-Mail is working, also via Bluetooth from device to device is working. I have tested it. This means there are alternatives. 

It is just the thing that it was working with V4 on this old device (I have used this very often), still working with V4 on the other devices with Android 14 and not with V5 what is the strange thing. 

When I would test again V4 on the old device, is there a possibility to backup the maps, relief data and pois from V5, then import it again in V4, just to see if it would still work then on the old device?  Or are these different formats?

I ask because I don't want to download again everything 2 times, takes very long.

What I also can do, I also have a Chinese outdoor device which I normally not use and could test if this happens also there with V5 together with WhatsApp.

**Reply by devemux86** [reply]
The thing is that it's too late to make comparisons with the previous version after so many months of beta testing.

There is no backwards compatibility in Android, only new versions can update old versions and use their data.

To make tests, it is not necessary to use offline maps, just install the app and start testing.
Once you decide to stay with a version, then you can download the offline data.

**Reply by DolomitesTom** [reply]
Ok thanks

**Comment by devemux86** [comment]
I tested Gmail attachments, Google Drive, Google Files and file managers and everything seems to work correctly.

**Reply by DolomitesTom** [reply]
With file manager it also works. What does also not work is when I open a Kurviger file from onedrive directly. Then I also get the message invalid file, but also even with V4.

**Reply by devemux86** [reply]
It works with Google Drive, so Microsoft's app may work differently.

The best solution is to use a file manager, like Google Files and click either the local file or the cloud file.

**Reply by DolomitesTom** [reply]
Sure, it was just a pretty easy and quick way to share via WhatsApp between devices and colleagues.

By the way, same error on the Oukitel device with V5 on Android 10.

**Comment by DolomitesTom** [comment]
Just for information: With K1 and K2, which is also from you, it is also working on my old devices with Android 10 when opening the Kurviger file as message in WhatsApp. I can also directly import it in K1 and K2 without using a file manager. It is not a device issue.

Sure one can live with alternatives, but It must be connected with Cruiser V5.

**Reply by devemux86** [reply]
This is very old code built with very old Android.
New Android does its own things with built apks.

**Comment by devemux86** [comment]
@DolomitesTom Try **5.0.1-b12**.

**Reply by DolomitesTom** [reply]
Tested b12: I think we get closer. It must be connected with the way how the Kurviger file is (exported)/ shared. When I share it with the V4 to WhatsApp, then it can be imported also in V5 directly without an issue. When I shared it with V5 then it cannot be imported again into V5.
The difference is when importing the V4 shared file, then the GPX Info is greyed out and it works 

![IMG-20241026-WA0009](https://github.com/user-attachments/assets/8ae10caf-c848-4c7b-9607-87a2c7661903)

importing the V5 shared file, then the GPX Info is in white text (active) and the error occurs.

![IMG-20241026-WA0010](https://github.com/user-attachments/assets/2948c199-559c-4b02-91e9-27ba6e6f6109)

**Reply by devemux86** [reply]
@DolomitesTom Thanks, this is good news, at least something was improved.

I will check more things. :slightly_smiling_face:

**Comment by devemux86** [comment]
@DolomitesTom Try **5.0.1-b13**.

**Reply by DolomitesTom** [reply]
Thanks for the update. Unfortunately still recognizes it as gpx file (not grayed out) and the message invalid file occurs.

**Reply by devemux86** [reply]
@DolomitesTom 4 to 5 works and 5 to 5 does not work?

This is strange, as the code is the same.

**Reply by DolomitesTom** [reply]
> 4 to 5 works and 5 to 5 does not work?

exactly and 5 to 4 works.  

one issue, just tested again: 4 to 5 works only when the tour is loaded fresh from my routes. When I load a tour from 5 to 4 and then directly share this imported route again to 5 without saving it in my routes, then also 4 to 5 doesn't work. Strange yes.

I don't understand it either. But this is exactly the behaviour and I can reproduce it with different routes.

**Reply by devemux86** [reply]
@DolomitesTom 

Please describe with details the steps you do and it works or not.
I do not understand with small descriptions what you mean.

**Comment by MotoP24** [comment]
Hello, the same issue during import directly from Whatsapp, also reported from my friends
Cruiser 5.0.0 from Playstore

Route shared from Cruiser 5.0.0 via Whatsapp result in invalid file message during automatic import in Cruiser 5.0.0

Route shared from Cruiser 4.0.2 via Whatsapp result in in correct automatic import in Cruiser 5.0.0

Route shared from older Kurviger 1.* App via Whatsapp result in correct automatic import in Cruiser 5.0.0

Route shared from 5.0.0 via Google Drive result in invalid file message during automatic import in Cruiser 5.0.0 but import it hiwever anyway correct

Thanks a lot !!!

**Reply by devemux86** [reply]
@MotoP24 Thanks for the feedback!

**Reply by MotoP24** [reply]
Thanks !!! I will try Cruiser 5.0.1 b11 as soon as possible

**Comment by devemux86** [comment]
When you all write share with WhatsApp, what do you mean exactly?

Use the share menu of Cruiser? Do something with WhatsApp app?

Can you show screenshots of your steps with WhatsApp to understand what you mean?

**Reply by MotoP24** [reply]
Exactly !!! Use the share menu of Cruiser 5.0.0 and them choise Whatsapp as target

**Reply by MotoP24** [reply]
I think the produced .kurviger file via share is not the issue but the automatic import of Cruiser 5.0.0 !!!

Because when I save the described shared .kurviger file from Whatsapp to my Samsung S20 FE local and then dont open the file but first start Cruiser 5.0.0 and import here the file manually from Cruiser menu, then it works !!!

**Reply by DolomitesTom** [reply]
Opening via file system works always. But this requires a lot of additional steps to get it to the other device. Via WhatsApp is much quicker

**Reply by devemux86** [reply]
@MotoP24 Open via files works.

Android detection of formats seems to not work properly, at least with some apps like WhatsApp.
Because Google Drive and Gmail work correctly too.

**Comment by DolomitesTom** [comment]
> Use the share menu of Cruiser?

Yes and then on the other device

Just tap on the shared message in WhatsApp
![Screenshot_20241027_201416_WhatsApp](https://github.com/user-attachments/assets/97cef039-3bf6-48b1-a3bc-1b8b50e3a82b)

And open with Cruiser

![Screenshot_20241027_201609_Android System](https://github.com/user-attachments/assets/db3837cf-514d-4ac7-afb1-9b5d8783dc1d)

**Reply by devemux86** [reply]
@DolomitesTom Thanks for the details!

**Comment by devemux86** [comment]
I have an idea how we could "force" the format, I need to make some tests.

**Comment by devemux86** [comment]
@DolomitesTom @MotoP24 Please try **Cruiser 5.0.1-b14**.

When we press a route in the file system or in another app like WhatsApp, the import dialog contains a format option.
If the format cannot be found, the last used format is selected. You can still change it if the format is not found correctly.

<img src="https://github.com/user-attachments/assets/8933ff66-2b70-42c6-bc67-7e88ef8f6810" width="50%"/>

**Reply by MotoP24** [reply]
sorry for the noob question... but where can I download this beta releses of Cruiser 5.0.1 ?

**Reply by devemux86** [reply]
@MotoP24 It is available [here](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0).

**Comment by DolomitesTom** [comment]
Top, this sorts it out, helps perfectly 💪👍

**Reply by MotoP24** [reply]
Yes !!! I can confirm it too - Cruiser 5.0.1 b14 works !!! Automatic import from Whatsapp or Google Drive without any issues or error messages.

**Reply by devemux86** [reply]
@DolomitesTom @MotoP24 

Very nice that we were able to solve it this way. Thank you both for your reports! :+1:

**Reply by MotoP24** [reply]
matter of honor !!! I love Cruiser and my moto buddies too

**Comment by devemux86** [comment, marked as answer]
- Fixed in https://github.com/devemux86/cruiser/discussions/641

**Comment by Leeb64** [comment]
Sharing and opening GPX and Kurviger files works with WhatsApp. It doesn't work with JSON files.
With SIGNAL it only works with GPX files.
But with SIGNAL it's not so bad because you can simply download the files.

**Reply by devemux86** [reply]
Note which format is selected in the import dialog, like in the [above image](https://github.com/devemux86/cruiser/discussions/651#discussioncomment-11074957).

(kurviger files are JSON files, they are the same)
